qiskit_nature/drivers/pyscfd/pyscfdriver.py

```diff
@@ -53,47 +68,79 @@ def __init__(
+        chkfile: Optional[str] = None,
         molecule: Optional[Molecule] = None,
     ) -> None:
         """
         Args:
-            atom: Atom list or string separated by semicolons or line breaks. Each element in the
-                list is an atom followed by position e.g. `H 0.0 0.0 0.5`. The preceding example
-                shows the `XYZ` format for position but `Z-Matrix` format is supported too here.
-            unit: Angstrom or Bohr
-            charge: Charge on the molecule
-            spin: Spin (2S), in accordance with how PySCF defines a molecule in pyscf.gto.mole.Mole
-            basis: Basis set name as recognized by PySCF, e.g. `sto3g`, `321g` etc.
-                See https://sunqm.github.io/pyscf/_modules/pyscf/gto/basis.html for a listing.
-                Defaults to the minimal basis 'sto3g'.
-            hf_method: Hartree-Fock Method type
-            conv_tol: Convergence tolerance see PySCF docs and pyscf/scf/hf.py
-            max_cycle: Max convergence cycles see PySCF docs and pyscf/scf/hf.py,
-                has a min. value of 1.
-            init_guess: See PySCF pyscf/scf/hf.py init_guess_by_minao/1e/atom methods
-            max_memory: Maximum memory that PySCF should use
-            molecule: A driver independent Molecule definition instance may be provided. When
+            atom: A string (or a list thereof) denoting the elements and coordinates of all atoms in
+                the system. Two formats are allowed; first, the PySCF-style `XYZ` format which is a
+                list of strings formatted as `{element symbol} {x_coord} {y_coord} {z_coord}`. If a
+                single string is given, the list entries should be joined by `;` as in the example:
+                `H 0.0 0.0 0.0; H 0.0 0.0 0.735`.
+                Second, the `Z-Matrix` format which is explained at 1_. The previous example
+                would be written as `H; H 3 0.735`.
+                See also 2_ for more details on geometry specifications supported by PySCF.
+            unit: Denotes the unit of coordinates. Valid values are given by the ``UnitsType`` enum.
+            charge: The charge of the molecule.
+            spin: The spin of the molecule. In accordance with PySCF's definition, the spin equals
+                :math:`2*S`, where :math:`S` is the total spin number of the molecule.
+            basis: A basis set name as recognized by PySCF (3_), e.g. `sto3g` (the default), `321g`,
+                etc. Note, that more advanced configuration options like a Dictionary or custom
+                basis sets are not allowed for the moment. Refer to 4_ for an extensive list of
+                PySCF's valid basis set names.
+            hf_method: The SCF method type to be used for the PySCF calculation. While the name
+                refers to HF methods, the PySCFDriver also supports KS methods. Refer to the
+                ``HFMethodType`` for a list of the supported methods.
```

Comment:
If the name needs this further explanation around hf would it be better to have this as a different name? Maybe simply just `method` that accommodates either and deprecate the `hf_method`; same for HFMethodType maybe it needs an enum specific to PySCF - that one supported  all the drivers and was used in PyQuante where these additional methods would be invalid.

Reply:
I agree that the naming is not ideal. Given our plans for the `MoleculeDriver` I think we can tackle that then, because we need to re-evaluate how this Enum ties into the general case there, too.